pkg/controller/nodes/task/k8s/plugin_manager.go

```diff
@@ -103,10 +106,37 @@ type PluginManager struct {
 	resourceLevelMonitor *ResourceLevelMonitor
 }
 
+// Handles dynamic templating of DefaultAnnotations and DefaultLabels to replace placeholders {{ PROJECT }} and {{ DOMAIN }}
+// with there actual values at execution time. This is useful if one wants to add custom IAM roles or Vault roles to task pods
+// depending on the project and domain.
+func InjectTemplates(taskCtx pluginsCore.TaskExecutionMetadata, defaults map[string]string) map[string]string {
+	execId := taskCtx.GetTaskExecutionID().GetID().NodeExecutionId.GetExecutionId()
+	project := execId.GetProject()
+	domain := execId.GetDomain()
+
+	y, err := yaml.Marshal(defaults)
```

Comment:
I would avoid marshaling and unmarshaling (even though admittedly it may make the code look cleaner)...

This code has been [done here](https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/core/template/template.go) already... let's just use that instead..